### PR TITLE
Warn users when eval is skipped

### PR DESCRIPTION
**Pull request by gadcam**
Fix for #620 & #293.

### CODE REVIEW

**Comment by rbgirshick** [comment]
@gadcam please rebase and I'll merge it.

**Comment by rbgirshick** [comment]
Hi @gadcam, can you format the warning messages so that the line length is <= 88 characters (80 chars + 10% slack)? Thanks.

**Comment by gadcam** [comment]
Hi @rbgirshick 
Now it should look better :) 80 characters not more, not less.

File: detectron/datasets/json_dataset_evaluator.py [hunk 1]
```diff
@@ -137,6 +141,10 @@ def evaluate_boxes(
     if json_dataset.name.find('test') == -1:
         coco_eval = _do_detection_eval(json_dataset, res_file, output_dir)
     else:
+        logger.warning(
+			'{} eval ignored as annotations are undisclosed on test: {} ignored'
```
Comment (rbgirshick) [inline detectron/datasets/json_dataset_evaluator.py]: Looks like there's extra indentation here.

Comment (gadcam) [reply detectron/datasets/json_dataset_evaluator.py]: @rbgirshick You are right, this line is tab indented.
I am really sorry about it, I double checked it should be mergeable now.

**Pull request closed without merging**